utils/readline/hotkey_functions.go

```diff
@@ -139,35 +139,58 @@ func HkFnCancelAction(rl *Instance) {
 	print(output)
 }
 
-func HkFnRecallWord1(rl *Instance)  { hkFnRecallWord(rl, 1) }
-func HkFnRecallWord2(rl *Instance)  { hkFnRecallWord(rl, 2) }
-func HkFnRecallWord3(rl *Instance)  { hkFnRecallWord(rl, 3) }
-func HkFnRecallWord4(rl *Instance)  { hkFnRecallWord(rl, 4) }
-func HkFnRecallWord5(rl *Instance)  { hkFnRecallWord(rl, 5) }
-func HkFnRecallWord6(rl *Instance)  { hkFnRecallWord(rl, 6) }
-func HkFnRecallWord7(rl *Instance)  { hkFnRecallWord(rl, 7) }
-func HkFnRecallWord8(rl *Instance)  { hkFnRecallWord(rl, 8) }
-func HkFnRecallWord9(rl *Instance)  { hkFnRecallWord(rl, 9) }
-func HkFnRecallWord10(rl *Instance) { hkFnRecallWord(rl, 10) }
-func HkFnRecallWord11(rl *Instance) { hkFnRecallWord(rl, 11) }
-func HkFnRecallWord12(rl *Instance) { hkFnRecallWord(rl, 12) }
+func HkFnRecallWord1(rl *Instance)    { hkFnRecallWord(rl, 1) }
+func HkFnRecallWord2(rl *Instance)    { hkFnRecallWord(rl, 2) }
+func HkFnRecallWord3(rl *Instance)    { hkFnRecallWord(rl, 3) }
+func HkFnRecallWord4(rl *Instance)    { hkFnRecallWord(rl, 4) }
+func HkFnRecallWord5(rl *Instance)    { hkFnRecallWord(rl, 5) }
+func HkFnRecallWord6(rl *Instance)    { hkFnRecallWord(rl, 6) }
+func HkFnRecallWord7(rl *Instance)    { hkFnRecallWord(rl, 7) }
+func HkFnRecallWord8(rl *Instance)    { hkFnRecallWord(rl, 8) }
+func HkFnRecallWord9(rl *Instance)    { hkFnRecallWord(rl, 9) }
+func HkFnRecallWord10(rl *Instance)   { hkFnRecallWord(rl, 10) }
+func HkFnRecallWord11(rl *Instance)   { hkFnRecallWord(rl, 11) }
+func HkFnRecallWord12(rl *Instance)   { hkFnRecallWord(rl, 12) }
+func HkFnRecallWordLast(rl *Instance) { hkFnRecallWordLast(rl) }
 
 const errCannotRecallWord = "Cannot recall word"
 
 func hkFnRecallWord(rl *Instance, i int) {
-	line, err := rl.History.GetLine(rl.History.Len() - 1)
-	if err != nil {
-		rl.ForceHintTextUpdate(fmt.Sprintf("%s %d: empty history", errCannotRecallWord, i))
+	tokens := getLastLineTokens(rl)
+	if tokens == nil {
+		return
+	} else if i > len(tokens) {
```

Comment:
No need for an else here


```suggestion
	}

if i > len(tokens) {
```